components/identity/identity-service.js

```diff
@@ -303,6 +303,13 @@ function factory($http) {
       }
       localStorage.setItem(
         STORAGE_KEYS.AUTHENTICATED, JSON.stringify(authenticated));
+
+      // update session
+      var session = service.getSession();
+      if(session) {
```

Comment:
No, I'm using the `id` that was passed in for what we're making permanent. If we don't have the same session as what we're making permanent, we shouldn't change its public key ID to the wrong one.